File: test-encryption.go
package main

import (
	"encoding/hex"
	"fmt"

	"github.com/unkeyed/unkey/pkg/encryption"
)

func main() {
	masterKey := "02a7a5b6c686c6c6f776f726c64"
	workspaceId := "ws_5ZUapZkYqhPpbkfL"
	testToken := "test_access_token_12345"

	fmt.Println("Testing encryption...")
	fmt.Println("MASTER_KEY length:", len(masterKey), "bytes")
	fmt.Println("MASTER_KEY bytes:", hex.EncodeToString([]byte(masterKey)))

	// Create encryption service
	enc, err := encryption.NewWorkspaceEncryption([]byte(masterKey))
	if err != nil {
		fmt.Printf("Error creating encryption: %v\n", err)
		return
	}

	// Encrypt
	encrypted, err := enc.EncryptToken(workspaceId, testToken)
	if err != nil {
		fmt.Printf("Error encrypting: %v\n", err)
		return
	}
	fmt.Println("Encrypted:", encrypted)

	// Decrypt
	decrypted, err := enc.DecryptToken(workspaceId, encrypted)
	if err != nil {
		fmt.Printf("Error decrypting: %v\n", err)
		return
	}
	fmt.Println("Decrypted:", decrypted)

	// Verify
	if decrypted == testToken {
		fmt.Println("✓ SUCCESS: Encryption/decryption works!")
	} else {
		fmt.Println("✗ FAILURE: Decrypted token doesn't match")
		fmt.Println("Expected:", testToken)
		fmt.Println("Got:", decrypted)
	}
}

File: test-encryption.ts
import { encryptToken, decryptToken } from "./web/apps/dashboard/lib/encryption";

// Test encryption/decryption
const masterKey = "02a7a5b6c686c6c6f776f726c64";
const workspaceId = "ws_5ZUapZkYqhPpbkfL";
const testToken = "test_access_token_12345";

console.log("Testing encryption...");
console.log("MASTER_KEY length:", masterKey.length, "bytes");
console.log("MASTER_KEY bytes:", Buffer.from(masterKey, "utf-8").toString("hex"));

// Encrypt
const encrypted = encryptToken(workspaceId, testToken, masterKey);
console.log("Encrypted:", encrypted);

// Decrypt
const decrypted = decryptToken(workspaceId, encrypted, masterKey);
console.log("Decrypted:", decrypted);

// Verify
if (decrypted === testToken) {
  console.log("✓ SUCCESS: Encryption/decryption works!");
} else {
  console.log("✗ FAILURE: Decrypted token doesn't match");
  console.log("Expected:", testToken);
  console.log("Got:", decrypted);
}